pkg/apiserver/archive/experiment.go

```diff
@@ -157,7 +212,44 @@ func (s *Service) experimentDetail(c *gin.Context) {
 		return
 	}
 
-	c.JSON(http.StatusOK, data)
+	switch data.Kind {
+	case v1alpha1.KindPodChaos:
+		info, err = data.ParsePodChaos()
+	case v1alpha1.KindIOChaos:
+		info, err = data.ParseIOChaos()
+	case v1alpha1.KindNetworkChaos:
+		info, err = data.ParseNetworkChaos()
+	case v1alpha1.KindTimeChaos:
+		info, err = data.ParseTimeChaos()
+	case v1alpha1.KindKernelChaos:
+		info, err = data.ParseKernelChaos()
+	case v1alpha1.KindStressChaos:
+		info, err = data.ParseStressChaos()
+	}
+	if err != nil {
+		c.Status(http.StatusInternalServerError)
+		_ = c.Error(utils.ErrInternalServer.WrapWithNoMessage(err))
+		return
+	}
+	expDetail = ArchiveExperimentDetail{
+		ArchiveExperimentMeta: core.ArchiveExperimentMeta{
+			ID:         data.ID,
+			CreatedAt:  data.CreatedAt,
+			UpdatedAt:  data.UpdatedAt,
+			DeletedAt:  data.DeletedAt,
+			Name:       data.Name,
+			Namespace:  data.Namespace,
+			Kind:       data.Kind,
+			Action:     data.Action,
+			UID:        data.UID,
+			StartTime:  data.StartTime,
+			FinishTime: data.FinishTime,
+			Archived:   data.Archived,
+		},
+		ExperimentInfo: info,
+	}
+
+	c.JSON(http.StatusOK, expDetail)
```

Comment:
Seems this part of code and above are dealing with the same thing, can we bump them into a common function?

Reply:
These two are not the same, one is to return a list of archives, and the other is to return a specific archive.

Reply:
Got it, my mistake. 🤣